test/spec/Feature/Query/SpreadQueriesSpec.hs

```diff
@@ -112,3 +90,513 @@ spec =
+            {"factory":"Factory A","name":["Process A1", "Process A2"],"category_id":[1, 2]},
+            {"factory":"Factory B","name":["Process B1", "Process B2"],"category_id":[1, 1]},
+            {"factory":"Factory C","name":["Process C1", "Process C2", "Process XX", "Process YY"],"category_id":[2, 2, 2, 2]},
+            {"factory":"Factory D","name":[],"category_id":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+        get "/factories?select=factory:name,...processes(processes:name,categories:category_id)&order=name" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory A","processes":["Process A1", "Process A2"],"categories":[1, 2]},
+            {"factory":"Factory B","processes":["Process B1", "Process B2"],"categories":[1, 1]},
+            {"factory":"Factory C","processes":["Process C1", "Process C2", "Process XX", "Process YY"],"categories":[2, 2, 2, 2]},
+            {"factory":"Factory D","processes":[],"categories":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should return an empty array when no elements are found" $
+        get "/factories?select=factory:name,...processes(processes:name)&processes=is.null" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory D","processes":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should return a single null element array, not an empty one, when the row exists but the value happens to be null" $
+        get "/managers?select=name,...organizations(organizations:name,referees:referee)&id=eq.1" `shouldRespondWith`
+          [json|[
+            {"name":"Referee Manager","organizations":["Referee Org"],"referees":[null]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should work when selecting all columns" $
+        get "/factories?select=factory:name,...processes(*)&order=name" `shouldRespondWith`
+          [json|[
+            {"factory":"Factory A","id":[1, 2],"name":["Process A1", "Process A2"],"factory_id":[1, 1],"category_id":[1, 2]},
+            {"factory":"Factory B","id":[3, 4],"name":["Process B1", "Process B2"],"factory_id":[2, 2],"category_id":[1, 1]},
+            {"factory":"Factory C","id":[5, 6, 7, 8],"name":["Process C1", "Process C2", "Process XX", "Process YY"],"factory_id":[3, 3, 3, 3],"category_id":[2, 2, 2, 2]},
+            {"factory":"Factory D","id":[],"name":[],"factory_id":[],"category_id":[]}
+          ]|]
+          { matchStatus  = 200
+          , matchHeaders = [matchContentTypeJson]
+          }
+      it "should select spread columns from a nested one-to-one relationship" $
+        get "/factories?select=factory:name,...processes(process:name,...process_costs(process_costs:cost))&order=name" `shouldRespondWith`
```

Comment:
Is this documented? This composability is interesting.

Reply:
No, not documented yet. Could be a subsection of to-many spreads and show an example there.